mesheryctl/internal/cli/root/pattern/export.go

```diff
@@ -0,0 +1,209 @@
+package pattern
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/server/handlers"
+	"github.com/layer5io/meshery/server/models"
+	"github.com/layer5io/meshkit/models/oci"
+	location "github.com/layer5io/meshkit/utils"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	save string
+	wd string
+)
+
+var exportCmd = &cobra.Command{
+	Use:   "export",
+	Short: "Export pattern",
+	Long:  `Export pattern to OCI format`,
+	Args:  cobra.MinimumNArgs(1),
+	Example: `
+// export a pattern to default path
+mesheryctl pattern export [pattern-name | ID]
+mesheryctl pattern export "my meshery design"
+
+// export pattern to the path that given by user
+mesheryctl pattern export [pattern-name | ID] -o [path]
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			utils.Log.Error(err)
+			return nil
+		}
+		pattern := ""
+		isID := false
+
+		if len(args) > 0 {
+			pattern, isID, err = utils.ValidId(mctlCfg.GetBaseMesheryURL(), args[0], "pattern")
+			if err != nil {
+				utils.Log.Error(ErrPatternInvalidNameOrID(err))
```

Comment:
Reminder: we will change from the word “pattern” to the word “design” very soon.

Reply:
I will change to design

Reply:
@Althaf66 good, although, only after the `app` to `pattern` transition is complete and is bug-free.